__tests__/integration/one-app.spec.js

```diff
@@ -962,6 +953,158 @@ describe('Tests that require Docker setup', () => {
+                    'franks-burgers.browser.js',
+                    'Burger.franks-burgers.chunk.browser.js'
+                  ),
+                ]
+              );
+            });
+          });
+
+          describe('cache invalidation', () => {
+            beforeAll(async () => {
+              await addModuleToModuleMap({
+                moduleName: 'late-frank',
+                version: '0.0.0',
+              });
+              // wait for change to be picked up
+              await waitFor(5000);
+            });
+
+            describe('module version', () => {
+              const fetchServiceWorkerScript = () => fetch(`${appAtTestUrls.fetchUrl}/_/pwa/service-worker.js`, {
+                ...defaultFetchOptions,
+              }).then((res) => res.text());
+
+              test('service worker script is updated when the module map changes with new module version', async () => {
+                expect.assertions(2);
+                // due to the service worker script changing and being re-installed
+                // selenium does not seem to update the service worker, we can check if the
+                // service worker script is updated to make sure the latest module would be updated
+                await expect(fetchServiceWorkerScript()).resolves.toContain('late-frank/0.0.0');
+                await addModuleToModuleMap({
+                  moduleName: 'late-frank',
+                  version: '0.0.1',
+                });
+                // wait for change to be picked up
+                await waitFor(5000);
+                await expect(fetchServiceWorkerScript()).resolves.toContain('late-frank/0.0.1');
+              });
+            });
+
+            describe('module language pack', () => {
+              test('invalidates a language pack if a different locale is loaded', async () => {
+                expect.assertions(2);
+
+                await browser.url(`${appAtTestUrls.browserUrl}/demo/cultured-frankie`);
+
+                const holocronModuleMap = readModuleMap();
+                const localeSelector = await browser.$('#locale-selector');
+                // update the locale
+                await localeSelector.selectByVisibleText('es-MX');
+                await waitFor(500);
+
+                const cacheKeys = await browser.executeAsync(getCacheKeys);
+                const cacheMap = new Map(await browser.executeAsync(getCacheEntries, cacheKeys));
+                // we should not expect an additional lang pack to be added,
+                // rather it replaces the other one used by any given module
+                expect(cacheMap.get('__sw/lang-packs')).toHaveLength(1);
+                expect(cacheMap.get('__sw/lang-packs')).toEqual(
+                  [
+                    holocronModuleMap.modules['cultured-frankie'].browser.url.replace(
+                      'cultured-frankie.browser.js',
+                      'es-mx/cultured-frankie.json'
+                    ),
+                  ]
```

Comment:
This replace calculation would be more readable on a separate line with its own variable.

Reply:
I have updated this (bf5db7c)